rust/src/mlil/function.rs

```diff
@@ -89,6 +96,21 @@ impl MediumLevelILFunction {
     }
 }
 
+impl ILFunction for MediumLevelILFunction {
+    type BNInstruction = BNMediumLevelILInstruction;
+    type Instruction = MediumLevelILInstruction;
+
+    fn il_instruction_from_idx(&self, expr_idx: usize) -> Self::Instruction {
+        self.instruction_from_idx(expr_idx)
+    }
+
+    fn operands_from_idx(&self, expr_idx: usize) -> [u64; 5] {
+        let node = unsafe { BNGetMediumLevelILByIndex(self.handle, expr_idx) };
+        assert_eq!(node.operation, BNMediumLevelILOperation::MLIL_UNDEF);
```

Comment:
Was just testing the `mlil_visitor` example and seems like this assertion ends up failing for some reason. I'll have to diagnose this and figure out what the behavioral difference here is.

Reply:
Figured it out - my iterator implementation was only accounting for lists of non-zero length.